### DISCUSSION TITLE
Problem about the physics plugin can't find body with DEF name

### DISCUSSION THREAD

**Discussion by izzm123** [cyberbotics/webots#4817] [Q&A]
Hi, I have created a physics plugin for the simulation world, and I used the function dWebotsGetBodyFromDEF( ). The solid nodes in the scene tree have been defined with DEF names. These names were used as the variables of the function dWebotsGetBodyFromDEF( ). The physics plugin has been added to the worldinfo.

As a consequence, when the world was run, the function dWebotsGetBodyFromDEF( ) returned  NULL. 

could you give me some suggestions?
Thanks!

**Comment by ad-daniel** [comment]
From the way you describe it, it should work. Can you provide a simple example of world/plugin that shows the issue?

**Reply by izzm123** [reply]
![image](https://user-images.githubusercontent.com/49048743/176842419-05b095f5-937a-4b4d-8391-0d27364bb931.png)
![image](https://user-images.githubusercontent.com/49048743/176842466-1cc29857-fd16-4d08-9f52-7149927ccd93.png)
![image](https://user-images.githubusercontent.com/49048743/176842502-a60012b4-f43b-4e6b-a393-9a2e95241277.png)

Thank you for taking time out of your busy schedule to help me with the issue. Above are some screenshots of the issue.

**Reply by izzm123** [reply]
[demo1.zip](https://github.com/cyberbotics/webots/files/9026885/demo1.zip)
Here is the source file, please.

**Reply by ad-daniel** [reply]
The issue is that you didn't define the `physics` field of your `Solid` and therefore the body doesn't exist on the physics engine side. Adding it should fix your issue

**Reply by izzm123** [reply]
Oh yeah, I found it also right when you replyed me back. Thanks! 非常感谢！给你点赞！
